addons/web_editor/static/src/js/editor/drag_and_drop.js

```diff
@@ -206,3 +208,381 @@ const dragAndDropHookParams = {
+        const currentHeight = this.dragState.currentHeight;
+        const backgroundGridEl = this.dragState.backgroundGridEl;
+        const dropzoneEl = this.dragState.currentDropzoneEl;
+        const rowOverflow = Math.round((bottom - currentHeight) / (gridProp.rowSize + gridProp.rowGap));
+        const updateRows = bottom > currentHeight || bottom <= currentHeight && bottom > startingHeight;
+        const rowCount = Math.max(rowEl.dataset.rowCount, this.dragState.columnRowCount);
+        const maxRowEnd = rowCount + gridUtils.additionalRowLimit + 1;
+        if (Math.abs(rowOverflow) >= 1 && updateRows) {
+            if (rowEnd <= maxRowEnd) {
+                const dropzoneEnd = parseInt(dropzoneEl.style.gridRowEnd);
+                dropzoneEl.style.gridRowEnd = dropzoneEnd + rowOverflow;
+                backgroundGridEl.style.gridRowEnd = dropzoneEnd + rowOverflow;
+                this.dragState.currentHeight += rowOverflow * (gridProp.rowSize + gridProp.rowGap);
+            } else {
+                // Don't add new rows if we have reached the limit
+                dropzoneEl.style.gridRowEnd = maxRowEnd;
+                backgroundGridEl.style.gridRowEnd = maxRowEnd;
+                this.dragState.currentHeight = (maxRowEnd - 1) * (gridProp.rowSize + gridProp.rowGap) - gridProp.rowGap;
+            }
+        }
+    }
+    /**
+     * Changes some behaviors before the drag and drop.
+     *
+     * @private
+     * @returns {Function} a function that restores what was changed when the
+     *  drag and drop is over.
+     */
+    prepareDrag() {
+        return () => {};
+    }
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Transforms the dragged item into a grid element if it is not one already.
+     *
+     * @private
+     * @param {HTMLElement} rowEl - The grid row.
+     */
+    _handleGridItemCreation(rowEl) {
+        if (!this.draggedItemEl.classList.contains("o_grid_item")) {
+            // Converting the column to grid
+            this.options.wysiwyg.odooEditor.observerActive("dragAndDropMoveSnippet");
+            const spans = gridUtils._convertColumnToGrid(rowEl, this.draggedItemEl, this.dragState.columnWidth, this.dragState.columnHeight);
+            this.options.wysiwyg.odooEditor.observerUnactive("dragAndDropMoveSnippet");
+
+            // Storing the column spans
+            this.dragState.columnColCount = spans.columnColCount;
+            this.dragState.columnRowCount = spans.columnRowCount;
+        }
+    }
+    /**
+     * Escapes the previous dropzone when an "over" dropzone event happens after
+     * an other "over" without an "out" between them.
+     *
+     * @private
+     * @param {HTMLElement} currentDropzoneEl - The dropzone over which we are
+     * currently dragging.
+     */
+    _outPreviousDropzone(currentDropzoneEl) {
```

Comment:
I actually think that this method is useless since [this commit](https://github.com/odoo/odoo/commit/7594d71ca8610d5947e80f325ccb57abc23c2c76) as the problem of an "over after an "over" does not seem to appear since it.